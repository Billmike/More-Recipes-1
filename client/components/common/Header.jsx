import React from 'react';

import Navbar from './Navbar';
import SideNav from './SideNav';

const Header = props => (
  <div>
    <div className="navbar-fixed">
      <nav className="deep-orange darken-4">
        <Navbar {...props} />
      </nav>
    </div>
    <SideNav />
  </div>
);

export default Header;
