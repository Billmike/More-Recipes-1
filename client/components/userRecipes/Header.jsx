import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import SideNav from '../common/SideNav';

/**
 * Class representing Header component
 *
 * @class Header
 * @extends {React.Component}
 */
class Header extends React.Component {
  /**
   * Creates an instance of Header.
   *
   * @memberof Header
   */
  constructor() {
    super();
    this.handleLogoutUser = this.handleLogoutUser.bind(this);
  }
  /**
   * Component did mount lifecycle method
   *
   * @returns {function} Initialize materialize componenets
   * @memberof Header
   */
  componentDidMount() {
    // Initialize materialize css side nav menu activator
    $('.button-collapse').sideNav({
      draggable: true,
    });
    // Initialize materialize dropdown class
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    // Initailize materialize tab class
    $('ul.tabs').tabs();
  }

  /**
   * Logout user from the application
   *
   * @returns {any} Logout user
   * @memberof Header
   */
  handleLogoutUser() {
    const { logoutUser, history } = this.props;
    logoutUser();
    history.push('/');
  }

  /**
   * Render method
   *
   * @returns {object} React element
   * @memberof Header
   */
  render() {
    return (
      <div>
        <div className="navbar-fixed">
          <nav className="nav-extended deep-orange darken-4">
            <div className="nav-wrapper" >
              <Link
                to="/"
                id="logo"
                className="brand-logo hide-on-small-only"
              >
                More-Recipes
              </Link>
              <a
                data-activates="slide_out"
                className="button-collapse"
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <div className="nav-content">
              <ul className="tabs tabs-transparent">
                <li className="tab"><a href="#user-profile">Profile</a></li>
                <li className="tab"><a href="#user-recipes">My Recipes</a></li>
                <li className="tab">
                  <a href="#user-favorites">My Favorites</a>
                </li>
              </ul>
            </div>
          </nav>
        </div>
        <SideNav {...this.props} />
      </div>
    );
  }
}

Header.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default Header;
