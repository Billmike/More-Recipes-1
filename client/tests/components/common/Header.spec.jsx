import React from 'react';

import Header from '../../../components/common/Header';

describe('Header component', () => {
  let spy;

  afterEach(() => {
    spy.mockClear();
  });
  it('should render component', () => {
    spy = jest.spyOn(Header.prototype, 'componentDidMount');
    const props = {
      signupUser: jest.fn(() => Promise.resolve()),
      signinUser: jest.fn(() => Promise.resolve()),
      logoutUser: jest.fn(() => Promise.resolve()),
      user: {
        isAuthenticated: false,
        userAuthentication: {
          message: null
        },
        error: {},
      },
      history: {
        push: jest.fn(),
      },
      location: {
        state: {
          from: {
            pathname: 'URL'
          },
        },
      },
      handleSearchCategory: jest.fn(),
    };
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toBeDefined();
    expect(spy).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component', () => {
    const props = {
      signupUser: jest.fn(() => Promise.resolve()),
      signinUser: jest.fn(() => Promise.resolve()),
      logoutUser: jest.fn(() => Promise.resolve()),
      user: {
        isAuthenticated: false,
        userAuthentication: {
          message: null
        },
        error: {},
      },
      history: {
        push: jest.fn(),
      },
      location: {
        state: {
          from: {
            pathname: 'URL'
          },
        },
      },
      handleSearchCategory: jest.fn(),
    };
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toBeDefined();
    wrapper.instance().handleToggleSignupModal();
    expect(wrapper.instance().state.openSignup).toEqual(true);
    wrapper.instance().handleToggleSigninModal();
    expect(wrapper.instance().state.openSignin).toEqual(true);
    wrapper.instance().handleToggleModal();
    expect(wrapper.instance().state.openSignup).toEqual(false);
    expect(wrapper.instance().state.openSignup).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle user signup event', () => {
    const props = {
      signupUser: jest.fn(() => Promise.resolve()),
      signinUser: jest.fn(() => Promise.resolve()),
      logoutUser: jest.fn(() => Promise.resolve()),
      user: {
        isAuthenticated: false,
        userAuthentication: {
          message: null
        },
        error: {},
      },
      history: {
        push: jest.fn(),
      },
      location: {
        state: {
          from: {
            pathname: 'URL'
          },
        },
      },
      handleSearchCategory: jest.fn(),
    };
    const wrapper = shallow(<Header {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleSubmitSignup');

    expect(wrapper).toBeDefined();
    wrapper.instance().handleSubmitSignup();
    expect(spy).toBeCalled();
    expect(props.signupUser).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle user signin event', () => {
    const props = {
      signupUser: jest.fn(() => Promise.resolve()),
      signinUser: jest.fn(() => Promise.resolve()),
      logoutUser: jest.fn(() => Promise.resolve()),
      user: {
        isAuthenticated: false,
        userAuthentication: {
          message: null
        },
        error: {},
      },
      history: {
        push: jest.fn(),
      },
      location: {
        state: {
          from: {
            pathname: 'URL'
          },
        },
      },
      handleSearchCategory: jest.fn(),
    };
    const wrapper = shallow(<Header {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleSubmitSignin');

    expect(wrapper).toBeDefined();
    wrapper.instance().handleSubmitSignin();
    expect(spy).toBeCalled();
    expect(props.signinUser).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle user signin and redirect user to former location', () => {
    const props = {
      signupUser: jest.fn(() => Promise.resolve()),
      signinUser: jest.fn(() => Promise.resolve()),
      logoutUser: jest.fn(() => Promise.resolve()),
      user: {
        isAuthenticated: false,
        userAuthentication: {
          message: null
        },
        error: {},
      },
      history: {
        push: jest.fn(),
      },
      location: {},
      handleSearchCategory: jest.fn(),
    };
    const wrapper = shallow(<Header {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleSubmitSignin');

    expect(wrapper).toBeDefined();
    wrapper.instance().handleSubmitSignin();
    expect(spy).toBeCalled();
    expect(props.signinUser).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle user logout event', () => {
    const props = {
      signupUser: jest.fn(() => Promise.resolve()),
      signinUser: jest.fn(() => Promise.resolve()),
      logoutUser: jest.fn(() => Promise.resolve()),
      user: {
        isAuthenticated: true,
        userAuthentication: {
          message: null
        },
        error: {},
      },
      history: {
        push: jest.fn(),
      },
      location: {
        state: {
          from: {
            pathname: 'URL'
          },
        },
      },
      handleSearchCategory: jest.fn(),
    };
    const wrapper = shallow(<Header {...props} />);
    spy = jest.spyOn(wrapper.instance(), 'handleLogoutUser');

    expect(wrapper).toBeDefined();
    wrapper.instance().handleLogoutUser();
    expect(spy).toBeCalled();
    expect(props.logoutUser).toBeCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
