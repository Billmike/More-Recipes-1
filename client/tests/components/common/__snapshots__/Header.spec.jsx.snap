// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should handle user logout event 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="deep-orange darken-4"
    >
      <Navbar
        handleLogoutUser={[Function]}
        handleSearchCategory={[MockFunction]}
        handleSubmitSignin={[Function]}
        handleSubmitSignup={[Function]}
        handleToggleModal={[Function]}
        handleToggleSigninModal={[Function]}
        handleToggleSignupModal={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "state": Object {
              "from": Object {
                "pathname": "URL",
              },
            },
          }
        }
        logoutUser={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
        openSignin={false}
        openSignup={false}
        signinUser={[MockFunction]}
        signupUser={[MockFunction]}
        user={
          Object {
            "error": Object {},
            "isAuthenticated": true,
            "userAuthentication": Object {
              "message": null,
            },
          }
        }
      />
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[Function]}
    handleSearchCategory={[MockFunction]}
    handleToggleSigninModal={[Function]}
    handleToggleSignupModal={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "state": Object {
          "from": Object {
            "pathname": "URL",
          },
        },
      }
    }
    logoutUser={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    openSignin={false}
    openSignup={false}
    signinUser={[MockFunction]}
    signupUser={[MockFunction]}
    user={
      Object {
        "error": Object {},
        "isAuthenticated": true,
        "userAuthentication": Object {
          "message": null,
        },
      }
    }
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</div>
`;

exports[`Header component should handle user signin and redirect user to former location 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="deep-orange darken-4"
    >
      <Navbar
        handleLogoutUser={[Function]}
        handleSearchCategory={[MockFunction]}
        handleSubmitSignin={[Function]}
        handleSubmitSignup={[Function]}
        handleToggleModal={[Function]}
        handleToggleSigninModal={[Function]}
        handleToggleSignupModal={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={Object {}}
        logoutUser={[MockFunction]}
        openSignin={false}
        openSignup={false}
        signinUser={
          [MockFunction] {
            "calls": Array [
              Array [
                undefined,
              ],
            ],
          }
        }
        signupUser={[MockFunction]}
        user={
          Object {
            "error": Object {},
            "isAuthenticated": false,
            "userAuthentication": Object {
              "message": null,
            },
          }
        }
      />
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[Function]}
    handleSearchCategory={[MockFunction]}
    handleToggleSigninModal={[Function]}
    handleToggleSignupModal={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    logoutUser={[MockFunction]}
    openSignin={false}
    openSignup={false}
    signinUser={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
      }
    }
    signupUser={[MockFunction]}
    user={
      Object {
        "error": Object {},
        "isAuthenticated": false,
        "userAuthentication": Object {
          "message": null,
        },
      }
    }
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</div>
`;

exports[`Header component should handle user signin event 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="deep-orange darken-4"
    >
      <Navbar
        handleLogoutUser={[Function]}
        handleSearchCategory={[MockFunction]}
        handleSubmitSignin={[Function]}
        handleSubmitSignup={[Function]}
        handleToggleModal={[Function]}
        handleToggleSigninModal={[Function]}
        handleToggleSignupModal={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "state": Object {
              "from": Object {
                "pathname": "URL",
              },
            },
          }
        }
        logoutUser={[MockFunction]}
        openSignin={false}
        openSignup={false}
        signinUser={
          [MockFunction] {
            "calls": Array [
              Array [
                undefined,
              ],
            ],
          }
        }
        signupUser={[MockFunction]}
        user={
          Object {
            "error": Object {},
            "isAuthenticated": false,
            "userAuthentication": Object {
              "message": null,
            },
          }
        }
      />
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[Function]}
    handleSearchCategory={[MockFunction]}
    handleToggleSigninModal={[Function]}
    handleToggleSignupModal={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "state": Object {
          "from": Object {
            "pathname": "URL",
          },
        },
      }
    }
    logoutUser={[MockFunction]}
    openSignin={false}
    openSignup={false}
    signinUser={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
      }
    }
    signupUser={[MockFunction]}
    user={
      Object {
        "error": Object {},
        "isAuthenticated": false,
        "userAuthentication": Object {
          "message": null,
        },
      }
    }
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</div>
`;

exports[`Header component should handle user signup event 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="deep-orange darken-4"
    >
      <Navbar
        handleLogoutUser={[Function]}
        handleSearchCategory={[MockFunction]}
        handleSubmitSignin={[Function]}
        handleSubmitSignup={[Function]}
        handleToggleModal={[Function]}
        handleToggleSigninModal={[Function]}
        handleToggleSignupModal={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "state": Object {
              "from": Object {
                "pathname": "URL",
              },
            },
          }
        }
        logoutUser={[MockFunction]}
        openSignin={false}
        openSignup={false}
        signinUser={[MockFunction]}
        signupUser={
          [MockFunction] {
            "calls": Array [
              Array [
                undefined,
              ],
            ],
          }
        }
        user={
          Object {
            "error": Object {},
            "isAuthenticated": false,
            "userAuthentication": Object {
              "message": null,
            },
          }
        }
      />
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[Function]}
    handleSearchCategory={[MockFunction]}
    handleToggleSigninModal={[Function]}
    handleToggleSignupModal={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "state": Object {
          "from": Object {
            "pathname": "URL",
          },
        },
      }
    }
    logoutUser={[MockFunction]}
    openSignin={false}
    openSignup={false}
    signinUser={[MockFunction]}
    signupUser={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
      }
    }
    user={
      Object {
        "error": Object {},
        "isAuthenticated": false,
        "userAuthentication": Object {
          "message": null,
        },
      }
    }
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</div>
`;

exports[`Header component should render component 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="deep-orange darken-4"
    >
      <Navbar
        handleLogoutUser={[Function]}
        handleSearchCategory={[MockFunction]}
        handleSubmitSignin={[Function]}
        handleSubmitSignup={[Function]}
        handleToggleModal={[Function]}
        handleToggleSigninModal={[Function]}
        handleToggleSignupModal={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "state": Object {
              "from": Object {
                "pathname": "URL",
              },
            },
          }
        }
        logoutUser={[MockFunction]}
        openSignin={false}
        openSignup={false}
        signinUser={[MockFunction]}
        signupUser={[MockFunction]}
        user={
          Object {
            "error": Object {},
            "isAuthenticated": false,
            "userAuthentication": Object {
              "message": null,
            },
          }
        }
      />
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[Function]}
    handleSearchCategory={[MockFunction]}
    handleToggleSigninModal={[Function]}
    handleToggleSignupModal={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "state": Object {
          "from": Object {
            "pathname": "URL",
          },
        },
      }
    }
    logoutUser={[MockFunction]}
    openSignin={false}
    openSignup={false}
    signinUser={[MockFunction]}
    signupUser={[MockFunction]}
    user={
      Object {
        "error": Object {},
        "isAuthenticated": false,
        "userAuthentication": Object {
          "message": null,
        },
      }
    }
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</div>
`;

exports[`Header component should render component 2`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="deep-orange darken-4"
    >
      <Navbar
        handleLogoutUser={[Function]}
        handleSearchCategory={[MockFunction]}
        handleSubmitSignin={[Function]}
        handleSubmitSignup={[Function]}
        handleToggleModal={[Function]}
        handleToggleSigninModal={[Function]}
        handleToggleSignupModal={[Function]}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        location={
          Object {
            "state": Object {
              "from": Object {
                "pathname": "URL",
              },
            },
          }
        }
        logoutUser={[MockFunction]}
        openSignin={false}
        openSignup={false}
        signinUser={[MockFunction]}
        signupUser={[MockFunction]}
        user={
          Object {
            "error": Object {},
            "isAuthenticated": false,
            "userAuthentication": Object {
              "message": null,
            },
          }
        }
      />
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[Function]}
    handleSearchCategory={[MockFunction]}
    handleToggleSigninModal={[Function]}
    handleToggleSignupModal={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "state": Object {
          "from": Object {
            "pathname": "URL",
          },
        },
      }
    }
    logoutUser={[MockFunction]}
    openSignin={false}
    openSignup={false}
    signinUser={[MockFunction]}
    signupUser={[MockFunction]}
    user={
      Object {
        "error": Object {},
        "isAuthenticated": false,
        "userAuthentication": Object {
          "message": null,
        },
      }
    }
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName=""
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName=""
    style={null}
    toastClassName=""
    transition={[Function]}
  />
</div>
`;
